core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -196,28 +196,44 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
       case NonFatal(e) => log.warn(e.getMessage, e)
     }
 
+    val hiveTables = Misc.getMemStore.getExternalCatalog.getHiveTables(true).asScala
```

Comment:
Should we wait for hive catalog initialization? What if hive is not initialized in lead yet?

Reply:
Probably not, as Hive catalog would have got initialized while creating SnappyContext.